### PR TITLE
WT-2372 - Fix windows inline define, changes for wt_internal inclusion

### PR DESCRIPTION
**Pull request by daveh86**

### CODE REVIEW

**Comment by daveh86** [comment]
@agorrod, would you mind taking a look at this?

File: src/include/msvc.h [hunk 1]
```diff
@@ -11,7 +11,9 @@
 #error "Only x64 is supported with MSVC"
 #endif
 
+#if !defined(__cplusplus) || !defined(inline)
```
Comment (agorrod) [inline src/include/msvc.h]: Do you need both of these? If so I'd recommend adding a comment to explain why.

File: ext/compressors/snappy/snappy_compress.c [hunk 2]
```diff
@@ -26,13 +26,13 @@
  * OTHER DEALINGS IN THE SOFTWARE.
  */
 
+#include <wt_internal.h>
```
Comment (agorrod) [inline ext/compressors/snappy/snappy_compress.c]: Why is this moved above the other includes? Generally it's better to include system headers first, so I don't really like this change.

**Comment by agorrod** [comment]
Thanks @daveh86 lgtm, I'll merge.
